mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -902,8 +902,16 @@ func resourceMongoDBAtlasClusterUpdate(ctx context.Context, d *schema.ResourceDa
 
 	// Has changes
 	if !reflect.DeepEqual(cluster, matlas.Cluster{}) {
-		err := resource.RetryContext(ctx, 3*time.Hour, func() *resource.RetryError {
-			_, _, err := updateCluster(ctx, conn, cluster, projectID, clusterName)
+		var err error
+		var updatedCluster *matlas.Cluster
+		isUpgrade := isUpgradeRequired(d)
+
+		err = resource.RetryContext(ctx, 3*time.Hour, func() *resource.RetryError {
+			if isUpgrade {
+				updatedCluster, _, err = upgradeCluster(ctx, conn, cluster, projectID, clusterName)
+			} else {
+				updatedCluster, _, err = updateCluster(ctx, conn, cluster, projectID, clusterName)
```

Comment:
this code has many nested ifs. Could you rewrite it in a way to limit the number of nested ifs?

Reply:
I reworked the un-pausing section to reduce nesting a bit, how's that?